dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java

```diff
@@ -618,19 +618,30 @@ public Iterable<SearchResult> call() throws Exception {
     private Task<Iterable<SearchResult>> getTaskForQueryDim(final String querySource, final String query,
             final DimLevel level, final Object... parameters) {
 
-        final QueryInterface queryEngine = getQueryProviderByName(querySource, true);
+        final QueryDimInterface queryEngine;
+        QueryDimInterface tmpQueryDimInterface;
+        try {
+            tmpQueryDimInterface = (QueryDimInterface) getQueryProviderByName(querySource, true);
+        } catch (NullPointerException ignored) {
+            tmpQueryDimInterface = null;
+        }
+
+        queryEngine = tmpQueryDimInterface;

```

Comment:
AFAIK a bad cast would throw `ClassCastException`, not `NullPointerException`. Best to just keep checking dynamically.

```suggestion
        final QueryDimInterface queryEngine;
        QueryInterface tmpQueryDimInterface = getQueryProviderByName(querySource, true);
        if (tmpQueryDimInterface instanceof QueryDimInterface) {
            queryEngine = (QueryDimInterface) tmpQueryDimInterface;
        } else {
            logger.warn("Query plugin {} is not a DIM query provider", querySource);
            queryEngine = null;
        }
```